node/shard/src/types.rs

```diff
@@ -3,17 +3,18 @@ use primitives::hash::{CryptoHash, hash_struct};
 use primitives::types::{
     AuthorityMask, MerkleHash, MultiSignature, PartialSignature, ShardId,
 };
-use transaction::Transaction;
+use primitives::traits::Payload;
+use transaction::{ReceiptTransaction, SignedTransaction};
 
-#[derive(Debug, Clone, PartialEq, Eq, Serialize, Deserialize)]
+#[derive(Hash, Debug, Clone, PartialEq, Eq, Serialize, Deserialize)]
```

Comment:
why do you need Hash here?

Reply:
Because `ReceiptBlock` needs to be hashable

Reply:
Implemented `Hash` in 00f85e7